File: generic_cpp/src/generic_tester_string_pub.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using namespace std::chrono_literals;

class GenericTesterStringPub : public rclcpp::Node
{
  public:
    GenericTesterStringPub(std::string topic_name)
      : Node("generic_tester_string_pub"), topic_name_(topic_name)
    {
      string_publisher_ = this->create_publisher<std_msgs::msg::String>(topic_name_, 10);

      timer_ = this->create_wall_timer(5s, std::bind(&GenericTesterStringPub::publish_messages, this));
    }
  private:
    void publish_messages()
    {
      std_msgs::msg::String string_msg;
      string_msg.data = "Generic subscriber working!";
      
      RCLCPP_INFO(this->get_logger(), "Publishing string message type of value: '%s'", string_msg.data.c_str());
      string_publisher_->publish(string_msg);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr string_publisher_;
    std::string topic_name_;
};

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);

  auto node = std::make_shared<GenericTesterStringPub>("/buffer");

  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: generic_py/generic_py/generic_tester_string_pub.py
from std_msgs.msg import String

import rclpy
from rclpy.node import Node

class GenericTesterStringPub(Node):

    def __init__(self, topic_name):
        super().__init__('generic_tester_string_pub')

        self.publisher_ = self.create_publisher(String, topic_name, 10)
        
        self.timer = self.create_timer(5.0, self.publish_messages)

    def publish_messages(self):
        msg = String()
        msg.data = "Generic subscriber working!"
        
        self.get_logger().info(f"Publishing string message type of value: '{msg.data}'")
        self.publisher_.publish(msg)


def main(args=None):
    rclpy.init(args=args)

    generic_tester_pub = GenericTesterStringPub("/buffer")

    rclpy.spin(generic_tester_pub)

    generic_tester_pub.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()